### ISSUE TITLE
Third party JS widget code is rendered with privacy settings on

### ISSUE THREAD

**Issue by Rudd-O** [collective/sc.social.like#81]
Somewhat related to #80.

For the expectation privacy to work, it isn't enough to just serve a link instead of the tracking button.  The page header must ALSO not link / import the JavaScript from the offending site (FB, Twitter, Google).

This isn't the case.  That means the browser, irrespective of DNT or site configuration, still attempts to fetch the JS from the offending site, slowing page loads and leaking data to the offending site.

A privacy feature implemented wrongly is worse than no privacy feature at all — it gives users and site operators the false sense of security that comes from thinking that their privacy is being respected, when in fact that isn't the case.

Please fix.

**Comment by Rudd-O** [comment]
Here is a sample of the code that the thing loads, irrespective of DNT setting:

```
  <script type="application/javascript">
    (function() {
        var po = document.createElement('script');
        po.async = true;
        po.src = document.location.protocol + '//connect.facebook.net/en_GB/all.js#xfbml=1';
        var head = document.getElementsByTagName('head')[0];
        head.appendChild(po);
    }());
    </script>

  
  
    
  <script type="application/javascript">
    (function() {
        var po = document.createElement('script');
        po.type = 'application/javascript';
        po.async = true;
        po.src = '//apis.google.com/js/plusone.js';
        var head = document.getElementsByTagName('head')[0];
        head.appendChild(po);
    })();
  </script>
```

The code appears to confirm this gets rendered irrespective of DNT or site settings.

**Comment by Rudd-O** [comment]
Sample from: http://www.brasil2016.gov.br/en/news/rio-2016-raises-the-bar-for-olympic-and-paralympic-sports

It's cool to have discovered that the Rio Brasil Olympics Web site has this sort of privacy hole.  I wonder who implemented it.

**Comment by hvelarde** [comment]
we implemented the site; thanks for your feedback and have a nice weekend.

CC @keul

**Comment by Rudd-O** [comment]
OK, excellent, that means I don't need to track them down and report the issue to them.

**Comment by hvelarde** [comment]
just to be clear: **there is no privacy hole in here**: the usage of [DNT request header](https://www.w3.org/TR/tracking-dnt/) is clear: a value of `1` means the user prefers not to be tracked **on the target site**; the target site is `www.brasil2016.gov.br` and we, in fact, are not tracking directly the user activity in anyway.

what @keul implemented takes a step further by removing the widgets; the obvious consequence is that we have sites that load faster and that's why we have that option enabled site wide there and not because of privacy concerns on the first place.

I don't know if Google, Facebook, Twitter, or other third parties respect that header or not (the current status is W3C Candidate Recommendation 20 August 2015) and that is outside the scope of this package and Plone in general.

I agree with you that is unnecessary to render the code above, but that's a different issue.

you have to be careful with claims like that.

**Comment by keul** [comment]
@Rudd-O nice catch. Yes this is a bug, including the JS is useless and probably harmful (although some of the URL triggered by JS manage no cookies at all... and at that time I only care about Facebook and Twitter), but I agree with @hvelarde: DNT is a *request* for not being tracked and I fear mostly ignored. You want real privacy? Go to anon mode through Tor.

We develop those changes for the EU Cookie Law but in the end this stayed in production only for few months because a second legal interpretation of the act reports that simply adding some few words in the privacy page was enough (although every EU country interpreted this in a differ manner).

I'm really sorry if this bug of mine hits you in any way but I can't be responsible of you issues in your country and life. See license: https://github.com/collective/sc.social.like/blob/master/docs/LICENSE.GPL

**Comment by Rudd-O** [comment]
I understand and I do not make you responsible for any issue stemming from trackers.  Just wanted users of this plugin to have the option to err on the side of more privacy.

**Comment by hvelarde** [comment]
partially addressed in #133.
